rfcs/0013-avalanche.md

```diff
@@ -0,0 +1,98 @@
+    | Validator 3 +------>| Validator 4 +══════>| Validator 5 +------>| Validator 6 |
+    |             |       |             |       |             |       |             |
+    `-------------`       `-------------`       `-------------`       `------+------`
+           ^                                                                 ║
+           ║                                                                 ║
+           ╚═════════════════════════════════════════════════════════════════╝
+
+#### What’s needed -
+The main goal of the new design is to enable Layer-2 nodes in the replication process. There are a few challenges with this and we should be able 
+to address some of them.  
+
+* Most importantly, there needs to be a mechanism to figure out which nodes belong in what layer of this avalanche. We need a Stake weighted selection for the nodes. This is missing and is very important. 
+* Leaders should only broadcast to layer 1 nodes and the network should be split (ClusterInfo can provide leader’s layer-1 tvu peers) 
+Layer 1 nodes should broadcast to all peers at the same layer and to a subset of peers at lower layers. 
+(ClusterInfo can provide Layer 1’s peers and layer 2 sections)
+* Layer 2 nodes should only retransmit amongst themselves. They should also be sectioned off and only need to communicate with each other instead of all layer 2 nodes. 
+(ClusterInfo can provide which nodes are in this node’s neighborhood)
+* 200 fanout from leader to layer-1 and layer-1 to layer-2.
+* Need to make sure layer-2 has a neighborhood (stake based) and that there can be some overlap. 
+
+
+### The proposed changes
+The idea is to fanout to 200 nodes from leader to layer 1 and then from layer 1 to layer 2. Layer 2 nodes should be semi-aware of this fanout so that they only talk to nodes in their "neighborhood". 
+This will result in the formation of multiple layer-2 neighborhoods that only care about communicating with each other and then the leader to send their votes. 
+Each layer 1 node only needs to broadcast its data to a single node in each neighborhood. With this in mind, the following changes will be needed to the two main stages.
+
+#### A Stake weighted selection mechanism:
+Need to come up with some logic to filter a set of nodes based on a stake range or number boundary. For the leader to find its layer 1 nodes, we need the biggest stake holders. 
+For each layer 1 node to find its layer 1 peers, the same logic *should* work but each layer 1 node also needs to find its layer 2 group. This needs some thought.
+Stake weights should work once we know the weight of the last layer 1 node. Layer 2 nodes can be "ordered" by weight and all layer 1 nodes should be able to avoid stepping over each others layer 2 nodes.    
+ 
+#### Broadcast stage:
+Broadcast Stage might need a bank to figure out stakes and can hand this off to ClusterInfo to figure out the top 200/85% stake holders. 
```

Comment:
I'm not sure this is needed. Anyone may validate. It's only the vote that's stake-weighted. I can see why we might put the highest stake nodes on layer 1 to minimize the network hops to achieve a supermajority vote, but that feels like a case of premature optimization. Let's assume that stake is spread out enough that some layer 2 nodes are needed to get to that supermajority. If that's the case, it doesn't matter if 1 or all heavily staked nodes are on layer 2. It only matters than the network latency of layer 1 nodes is not too high.